cpp/include/cudf/detail/utilities/hash_functions.cuh

```diff
@@ -430,7 +433,23 @@ template <>
 hash_value_type CUDA_DEVICE_CALLABLE
 SparkMurmurHash3_32<numeric::decimal128>::operator()(numeric::decimal128 const& key) const
 {
-  return this->compute<__int128_t>(key.value());
+  __int128_t const val    = key.value();
+  __int128_t flipped      = 0;
+  int8_t const* data      = reinterpret_cast<int8_t const*>(&val);
+  int8_t* dflipped        = reinterpret_cast<int8_t*>(&flipped);
+  int8_t const sign_bit   = data[15] & 0x80;
+  int8_t const zero_value = sign_bit ? 0xff : 0x00;
+  int32_t length          = 15;
+
+  // Remove preceeding zero values and flip endianess before hashing
```

Comment:
By preceding I mean data that can be removed without changing the fundamental value being represented.

e.g. a 8-byte value that can be represented as a 5 byte value. `0x000000FFFFFFFFFF` is reduced to `0x00FFFFFFFFFF` and `0xFFFFFFFF76E8C900` is reduced to `0xFF76E8C900` because it's necessary to retain the sign bit. In simpler cases `0x0000007FFFFFFFFF` is reduced to `0x7FFFFFFFFF` and `0xFFFFFFFF86E8C900` is reduced to `0x86E8C900`.